oauth-server-plain/src/main/java/io/strimzi/kafka/oauth/server/plain/JaasServerOauthOverPlainValidatorCallbackHandler.java

```diff
@@ -246,7 +246,7 @@ private void authenticate(String username, String password) throws UnsupportedCa
                 checkUsernameMatch = true;
             } else if (tokenEndpointUri != null) {
                 accessToken = OAuthAuthenticator.loginWithClientSecret(tokenEndpointUri, getSocketFactory(), getVerifier(),
-                        username, password, isJwt(), getPrincipalExtractor(), scope, audience, getConnectTimeout(), getReadTimeout(), authMetrics)
+                        username, password, isJwt(), null, null, getPrincipalExtractor(), scope, audience, getConnectTimeout(), getReadTimeout(), authMetrics)
```

Comment:
how would it look like?
the only reason why I added the `null, null`, is because the signature of `OAuthAuthenticator.loginWithClientSecret()` had changed to support passing the `username` and `password` fields.
what was not clear for me was that the `clientId` and `clientSecret` were taken from `void authenticate(String username, String password)`. what are these? a username of a client id? a password or a client secret?
If I pass values for the new `loginWithClientSecret(..., String username, String password)` params, what should be the values?

Reply:
It may be a little confusing yes. Here things are named from the point of view of SASL_PLAIN mechanism which propagates from the client two parameters called `username` and `password` because SASL_PLAIN in combination with Simple ACL authorization expects user to authenticate with the username and a password. We use these two attributes to pass different information from client to broker. For us username is either a username or a clientId, and password is either the client secret or an access token. While the variable is called username / password it contains something else.
It is not obvious how to add support for tunneling of username and password (in addition to clientId and a secret). There is only room for two attributes. Passing through potentially 4 pieces of information and encoding them into two is not obvious. I was more thinking aloud up there rather than wanting you to add this support.